src/extendedcalendar.cpp

```diff
@@ -40,93 +40,23 @@
 #include "sqlitestorage.h"
 #include "logging_p.h"
 
-#include <KCalendarCore/CalFilter>
 #include <KCalendarCore/CalFormat>
-#include <KCalendarCore/Sorting>
 using namespace KCalendarCore;
 
-#include <cmath>
-
-// #ifdef to control expensive/spammy debug stmts
-#undef DEBUG_EXPANSION
-
 using namespace mKCal;
-/**
-  Private class that helps to provide binary compatibility between releases.
-  @internal
-*/
-//@cond PRIVATE
-
-/**
-  Make a QHash::value that returns a QVector.
-*/
-template <typename K, typename V>
-QVector<V> values(const QMultiHash<K, V> &c, const K &x)
-{
-    QVector<V> v;
-    typename QMultiHash<K, V>::const_iterator it = c.find(x);
-    while (it != c.end() && it.key() == x) {
-        v.push_back(it.value());
-        ++it;
-    }
-    return v;
-}
-
-class mKCal::ExtendedCalendar::Private
-{
-public:
-    Private()
-    {
-    }
-    ~Private()
-    {
-    }
-    Incidence::List mGeoIncidences;                  // list of all Geo Incidences
-    QMultiHash<QString, Incidence::Ptr>mAttendeeIncidences; // lists of incidences for attendees
-
-    void addIncidenceToLists(const Incidence::Ptr &incidence);
-    void removeIncidenceFromLists(const Incidence::Ptr &incidence);
-
-    /**
-     * Figure when particular recurrence of an incidence starts.
-     *
-     * The start is a hint that is used as rough approximation (it may
-     * be result of expandMultiDay).
-     */
-    static QDateTime incidenceRecurrenceStart(const KCalendarCore::Incidence::Ptr &incidence, const QDateTime start);
-
-    /**
-     * Figure appropriate end time for incidence.
-     *
-     * This functon should be used only on results of expandMultiDay,
-     * and therefore the start may or may not be incidence start time.
-     *
-     * param start The start time gotten from say, expandMultiDay
-     * param endWithinDay Whether we should try to end within the day or not.
-     *
-     * return The time when the event ends. While this function returns
-     * QDateTime, the date will be always same as start.date(); this is
-     * just convenience API. Only exception is all-day case, where date
-     * = start.date()+1, and time is 0,0,0.
-     */
-    static QDateTime incidenceEndTime(const KCalendarCore::Incidence::Ptr &incidence,
-                                      const QDateTime start,
-                                      bool endWithinDay);
-};
 
 ExtendedCalendar::ExtendedCalendar(const QTimeZone &timeZone)
-    : MemoryCalendar(timeZone), d(new mKCal::ExtendedCalendar::Private)
+    : MemoryCalendar(timeZone)
 {
 }
 
 ExtendedCalendar::ExtendedCalendar(const QByteArray &timeZoneId)
-    : MemoryCalendar(timeZoneId), d(new mKCal::ExtendedCalendar::Private)
```

Comment:
I'd keep the private class even if it ended up empty. Could allow avoiding ABI break then some day when we need to store some member variable.

Reply:
Sure, I've reverted this change.